src/scheduler/server_info.c

```diff
@@ -1420,6 +1420,38 @@ create_resource(char *name, char *value, enum resource_fields field)
 	return nres;
 }
 
+int
+modify_resource_list(schd_resource *res_list, resource_req *req_list, int type)
+{
+	schd_resource *cur_res;
+	resource_req *cur_req;
+	schd_resource *end_res = NULL;
+
+	if (res_list == NULL || req_list == NULL)
+		return 0;
+	
+	for(cur_req = req_list; cur_req != NULL; cur_req = cur_req->next) {
+		if (cur_req->type.is_consumable) {
+			cur_res = find_resource(res_list, cur_req->def);
+			if (cur_res == NULL) {
+				if (end_res == NULL)
+					for (end_res = res_list; end_res->next != NULL; end_res = end_res->next)
+						;
+				end_res->next = create_resource(cur_req->name, cur_req->res_str, RF_AVAIL);
```

Comment:
shouldn't you check here whether 'type' is SCHD_INCR and only add to the list if true?